examples/hosts/app-integration/external-controller/src/app.ts

```diff
@@ -23,16 +22,18 @@ document.title = containerId;
 const dataObjectId = "dice";
 
 async function start(): Promise<void> {
-    const service = new TinyliciousService();
+    const containerConfig: FluidCreateContainerConfig = {
```

Comment:
I worry this is adding +1 concept for new developers to learn (`FluidCreateContainerConfig`, its format, what it actually does).  At best, I think we would have to avoid teaching `createDataObject` in order to call it a 1:1 concept swap, but I feel that the question of creating another data object will come up quickly (especially if we are talking about adding handle support to kvpair).

Reply:
I agree it's a new concept but I actually think it can be used as a learning tool. There are a few things I think the config gives us over the current flow. 

1. Defining a config allows for a singular object to be used for container creation as well as retrieval. This consolidates the config of the container in one location. No need for duplicate declaration of DataObjects.

2. By doing DO create as part of container create it ensure that all DO exist in the original snapshot of the container. This can prevent race conditions.

3. create specific logic is consolidated in one location so the developer doesn't need to manage DO initialization outside of the the container creation flow. This makes the initial DO pattern more explicit.

4. It standardizes initial DO creation into a schema. This makes defining multiple initial DO easier.

5. IMO the config object makes it easier to understand what I need to provide to the container. by saying {dataObjects: [Foo,Bar]} I'm clearly seeing that the dataObjects in this config are Foo and Bar. By passing the same config to create and get we show that you need to ensure the container config remains constant. This presents a good learning opportunity.

6. As we introduce new functionality and expand what the developer can control in the container we can extend the config with optional properties without changing the method signature. This will be valuable.

Reply:
FWIW I was excited when I saw the word config. May just be me, but config is familiar. I (think I) know what to expect.